crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3626,51 +3695,59 @@ pub async fn list_customer_payment_method(
 
         let intent_created = payment_intent.as_ref().map(|intent| intent.created_at);
 
-        let redis_conn = state
-            .store
-            .get_redis_conn()
-            .change_context(errors::ApiErrorResponse::InternalServerError)
-            .attach_printable("Failed to get redis connection")?;
-        ParentPaymentMethodToken::create_key_for_token((
-            &parent_payment_method_token,
-            pma.payment_method,
-        ))
-        .insert(
-            intent_created,
-            payment_method_retrieval_context.hyperswitch_token_data,
-            state,
-        )
-        .await?;
+        if is_payment_associated {
+            let token = parent_payment_method_token
+                .as_ref()
+                .get_required_value("parent_payment_method_token")?;
+            let hyperswitch_token_data = payment_method_retrieval_context
+                .hyperswitch_token_data
+                .as_ref()
+                .cloned()
+                .get_required_value("PaymentTokenData")?;
 
-        if let Some(metadata) = pma.metadata {
-            let pm_metadata_vec: payment_methods::PaymentMethodMetadata = metadata
-                .parse_value("PaymentMethodMetadata")
-                .change_context(errors::ApiErrorResponse::InternalServerError)
-                .attach_printable(
-                    "Failed to deserialize metadata to PaymentmethodMetadata struct",
-                )?;
+            ParentPaymentMethodToken::create_key_for_token((token, pma.payment_method))
+                .insert(intent_created, hyperswitch_token_data, state)
+                .await?;
+        }
 
-            for pm_metadata in pm_metadata_vec.payment_method_tokenization {
-                let key = format!(
-                    "pm_token_{}_{}_{}",
-                    parent_payment_method_token, pma.payment_method, pm_metadata.0
-                );
-                let current_datetime_utc = common_utils::date_time::now();
-                let time_elapsed = current_datetime_utc
-                    - payment_intent
-                        .as_ref()
-                        .map(|intent| intent.created_at)
-                        .unwrap_or_else(|| current_datetime_utc);
-                redis_conn
-                    .set_key_with_expiry(
-                        &key,
-                        pm_metadata.1,
-                        consts::TOKEN_TTL - time_elapsed.whole_seconds(),
-                    )
-                    .await
-                    .change_context(errors::StorageError::KVError)
+        // Check this block in review
+        if let Some(token) = parent_payment_method_token {
```

Comment:
We can just merge this block with the `if is_payment_associated {}` block right above right? `is_payment_associated` and `parent_payment_method_token.is_some()` should always have the same truth value.

Reply:
yes, they would have the same truth value, I wasn't sure what the block is supposed to do hence the comment. 
I'll merge this with the above block.